StripeFinancialConnections/StripeFinancialConnections/Source/Native/NativeFlowController.swift

```diff
@@ -445,6 +477,62 @@ extension NativeFlowController {
             }
     }
 
+    private func createPaymentMethod(
+        for session: StripeAPI.FinancialConnectionsSession,
+        completion: @escaping (Result<InstantDebitsLinkedBank, Error>) -> Void
+    ) {
+        let bankAccountId: String?
+        switch session.paymentAccount {
+        case .bankAccount(let account):
+            bankAccountId = account.id
+        case .linkedAccount(let account):
+            bankAccountId = account.id
+        default:
+            bankAccountId = nil
+        }
+
+        guard let bankAccountId else {
+            let error = "InstantDebitsCompletionError: No bank account ID available when trying to create a payment method."
+            completion(.failure(FinancialConnectionsSheetError.unknown(debugDescription: error)))
+            return
+        }
+
+        guard let consumerSession = dataManager.consumerSession else {
+            let error = "InstantDebitsCompletionError: No consumer session available when trying to create a payment method."
+            completion(.failure(FinancialConnectionsSheetError.unknown(debugDescription: error)))
+            return
+        }
+
+        var bankAccountDetails: BankAccountDetails?
+        dataManager.createPaymentDetails(
+            consumerSessionClientSecret: consumerSession.clientSecret,
+            bankAccountId: bankAccountId
+        ).chained { [weak self] paymentDetails -> Future<FinancialConnectionsPaymentMethod> in
+            guard let self else {
+                return Promise(error: FinancialConnectionsSheetError.unknown(debugDescription: "data source deallocated"))
+            }
+
+            bankAccountDetails = paymentDetails.redactedPaymentDetails.bankAccountDetails
```

Comment:
this is fine but also maybe possible to "chain" and "create new values/tuples" to avoid the weird `var`

https://github.com/stripe/stripe-ios/blob/7efd4201/StripeFinancialConnections/StripeFinancialConnections/Source/API%20Bindings/FinancialConnectionsAPIClient.swift#L631-L638

Reply:
I tried this at first, but wasn't sure how to cary over the bank account details into the next `chained` block :/